internal/distro/rhel7/pipelines.go

```diff
@@ -0,0 +1,335 @@
+package rhel7
+
+import (
+	"fmt"
+
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/disk"
+	"github.com/osbuild/osbuild-composer/internal/distro"
+	osbuild "github.com/osbuild/osbuild-composer/internal/osbuild2"
+	"github.com/osbuild/osbuild-composer/internal/rpmmd"
+)
+
+func buildPipeline(repos []rpmmd.RepoConfig, buildPackageSpecs []rpmmd.PackageSpec, runner string) *osbuild.Pipeline {
+	p := new(osbuild.Pipeline)
+	p.Name = "build"
+	p.Runner = runner
+	p.AddStage(osbuild.NewRPMStage(rpmStageOptions(repos), osbuild.NewRpmStageSourceFilesInputs(buildPackageSpecs)))
+	p.AddStage(osbuild.NewSELinuxStage(selinuxStageOptions(true)))
+	return p
+}
+
+func prependKernelCmdlineStage(pipeline *osbuild.Pipeline, t *imageType, pt *disk.PartitionTable) *osbuild.Pipeline {
+	if t.arch.name == distro.S390xArchName {
+		rootFs := pt.FindMountable("/")
+		if rootFs == nil {
+			panic("s390x image must have a root filesystem, this is a programming error")
+		}
+		kernelStage := osbuild.NewKernelCmdlineStage(osbuild.NewKernelCmdlineStageOptions(rootFs.GetFSSpec().UUID, t.kernelOptions))
+		pipeline.Stages = append([]*osbuild.Stage{kernelStage}, pipeline.Stages...)
+	}
+	return pipeline
+}
+
+func osPipeline(t *imageType,
+	repos []rpmmd.RepoConfig,
+	packages []rpmmd.PackageSpec,
+	c *blueprint.Customizations,
+	options distro.ImageOptions,
+	pt *disk.PartitionTable) (*osbuild.Pipeline, error) {
+
+	imageConfig := t.getDefaultImageConfig()
+	p := new(osbuild.Pipeline)
+
+	p.Name = "os"
+	p.Build = "name:build"
+
+	rpmOptions := osbuild.NewRPMStageOptions(repos)
+	rpmOptions.GPGKeysFromTree = imageConfig.GPGKeyFiles
+	p.AddStage(osbuild.NewRPMStage(rpmOptions, osbuild.NewRpmStageSourceFilesInputs(packages)))
+
+	// Difference to RHEL8, 9 pipelines: no BLS stage
```

Comment:
hmm... I wonder how could we keep the `osPipeline` implementations identical across distro definitions (to allow deduplicating it eventually), but still being able to not apply the BLS stage on RHEL-7 🤔 

What about adding `bool` variable to the `ImageConfig` to distinguish whether BLS should be used or not and making it part of the default distro `ImageConfig`?

Reply:
I think we can keep it right now as is, especially since it has this difference, and we can sort it out in the eventual deduplication, rather than adding dead code.